deploy-board/deploy_board/webapp/auth.py

```diff
@@ -224,20 +227,21 @@ def handle_oauth2_response(self, code, state_with_data, **kwargs):
         self.oauth_handler.token_setter(resp_data['access_token'], expires, **kwargs)
 
         try:
-            return json.loads(enc_data.decode('base64'))
+            return json.loads(enc_data)
         except ValueError:
             return None
 
     def get_authorization_url(self, data=None, **kwargs):
         client = self.get_client()
-        scope = unicode(self.scope)
+        scope = str(self.scope)
 
         # generate and set state
         state = self.oauth_handler.state_generator(**kwargs)
         self.oauth_handler.state_setter(state, **kwargs)
 
         # hack to add data to state
-        state_with_data = state + json.dumps(data).encode('base64')
+        encoded_data = base64.b64encode(json.dumps(data).encode('utf-8')).decode('utf-8')  
```

Comment:
This is decoded as per `.decode('utf-8')`, no? Variable is called `encoded_data`. Not sure which one is the intended

Reply:
encode with base64 and convert to a `str`.